src/if_checks.c

```diff
@@ -237,3 +237,40 @@ struct check_result *check_type_required_but_not_used_in_if(__attribute__((unuse
 	free_string_list(types_to_check);
 	return res;
 }
+
+struct check_result *check_if_calls_template(__attribute__((unused)) const struct
+                                             check_data *data,
+                                             const struct
+                                             policy_node *node)
+{
+	const struct policy_node *parent = node->parent;
+	while (parent &&
+	       (parent->flavor == NODE_OPTIONAL_POLICY || parent->flavor == NODE_IFDEF || parent->flavor == NODE_TUNABLE_POLICY)) {
+		parent = parent->parent;
+	}
+	
+	if (!parent) {
+		return NULL;
+	}
+	
+	const struct if_call_data *id = node->data.ic_data;
+	
+	if (parent->flavor != NODE_INTERFACE_DEF && parent->flavor != NODE_TEMP_DEF) {
```

Comment:
This case to me feels more like an internal error.

The most common case seems to be if we are in a block not listed above, which is likely an internal error because we should have listed the block above.

Why not just go up the parents until you find NODE_INTERFACE_DEF or NODE_TEMP_DEF?  If you don't, return NULL.  If you do, proceed.  That gets this check out of the business of listing all the blocks above on line 248.

Reply:
re-worked